spectral_cube/lower_dimensional_structures.py

```diff
@@ -387,3 +389,100 @@ def hdulist(self):
         beamhdu = beams_to_bintable(self.beams)
 
         return HDUList([hdu, beamhdu])
+
+
+    def spectral_interpolate(self, spectral_grid,
+                             suppress_smooth_warning=False):
+        """
+        Resample the spectrum onto a specific grid
+
+        Parameters
+        ----------
+        spectral_grid : array
+            An array of the spectral positions to regrid onto
```

Comment:
I wouldn't call it regridding, as that means something different to me than interpolation